code/modules/surgery/bodyparts/bodypart_effects.dm

```diff
@@ -0,0 +1,121 @@
+/// For scaling the effectiveness of certain effects to the total bodypart count
+#define GET_BODYPART_COEFFICIENT(X) round(X.len / 6, 0.1)
+/// Check if it's full body. These are mostly here so we can change just one place when we ever add more limbs (?)
+#define IS_FULL_BODY(X) (X.len == 6)
+
+/// Effects added to a carbon focused on the bodyparts itself, such as adding a photosynthesis component that
+/datum/bodypart_effect
+	/// List of bodyparts contributing to this effect
+	var/list/bodyparts = list()
+	/// Minimum amount of bodyparts required for on_apply to be called. When tipping below, on_remove is called
+	var/minimum_bodyparts = 1
+	/// Are we currently active? We don't NEED to track it, but it's a lot easier and faster if we do
+	var/is_active = FALSE
```

Comment:
This is currently unused, along with activate() and deactivate(), but ideally we could also use this to have some species traits appear when a certain threshold of limbs is reached. Something like 3 snail limbs, and now you can snailcrawl! (This is not here yet because we'd also need to modularise how we calculate stuff like speed)